src/stringProcessing/checkNofollow.js

```diff
@@ -1,17 +1,38 @@
 /** @module stringProcessing/checkNofollow */
 
+// We use an external library, which can be found here: https://github.com/fb55/htmlparser2.
+let htmlparser = require( "htmlparser2" );
+let isString = require( "lodash/isString" );
+
 /**
  * Checks if a links has a nofollow attribute. If it has, returns Nofollow, otherwise Dofollow.
  *
  * @param {string} text The text to check against.
  * @returns {string} Returns Dofollow or Nofollow.
  */
 module.exports = function( text ) {
-	var linkFollow = "Dofollow";
+	let linkFollow = "Dofollow";
+
+	let parser = new htmlparser.Parser( {
+		/**
+		 * Detects if there is a `nofollow` argument in the `rel` attribute of a link.
+		 *
+		 * @param {string} tagName The tag name.
+		 * @param {object} nodeValue The attribute with the keys and values of the tag.
+		 * @returns {void}
+		 */
+		onopentag: function( tagName, nodeValue ) {
+			if ( tagName.toLowerCase() !== "a" || ! isString( nodeValue.rel ) ) {
+				return;
+			}
+
+			if ( nodeValue.rel.toLowerCase().split( /\s/ ).includes( "nofollow" ) ) {
+				linkFollow = "Nofollow";
+			}
+		},
+	} );
+
+	parser.write( text );
```

Comment:
Not sure if we should also use `parser.end();`, as shown in all the examples in the htmlparser2 docs.